### PR TITLE
Add pfn2lfn function to the policy package

### PR DESCRIPTION
**Pull request by haozturk**
This is tested in integration and works as expected

### CODE REVIEW

File: src/policy/CMSRucioPolicy/algorithms/pfn2lfn.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+from collections.abc import Mapping
+
+def cms_pfn2lfn(parsed_pfn: Mapping[str, str]) -> tuple[str, str]:
+    """
+    This function converts a PFN into scope:name pair.
+    It ignores user scopes and sets all scopes to cms.
+    """
+    # Ignore user scopes for now
+    scope = "cms"
+    name = parsed_pfn['path'] + parsed_pfn['name']
```
Comment (dynamic-entropy) [inline src/policy/CMSRucioPolicy/algorithms/pfn2lfn.py:13]: Hmm, I do not seem to understand this. 
The `lfn` would only be "name" and not a concat of `path` and `name`.

**Review by dynamic-entropy — CHANGES_REQUESTED** [review]
What all info does the "parsed_pfn" dict contain? 
And do we know how it gets generated? Seems different for different protocols. 

Would this work for sites with `cmstfc` lfn2pfn algo?

File: src/policy/CMSRucioPolicy/algorithms/pfn2lfn.py (hunk 1 above)
Comment (haozturk) [reply src/policy/CMSRucioPolicy/algorithms/pfn2lfn.py:13]: Not really, this is what `parsed_pfn` looks like for an example cms PFN:
```
{'scheme': 'davs', 'port': '1094', 'hostname': 'cmsxrootd.hep.wisc.edu', 'path': '/store/backfill/1/CMSSW_12_3_0/DoubleMuon_New/FEVTDEBUGHLT/HLTDR2_2018_SC_LumiMask_Rules_June2024_Val_v1-v11/00000/', 'name': 'e6f95196-5df4-4d03-a786-2fd3fcb968b1.root', 'prefix': '/store/temp/sitetest', 'web_service_path': ''
...
```

Comment (dynamic-entropy) [reply src/policy/CMSRucioPolicy/algorithms/pfn2lfn.py:13]: Ok perfect!

**Comment by dynamic-entropy** [comment]
@haozturk 
Can you create this PR against the `rucio_35` branch please incase you want to deploy it to prod?

**Comment by haozturk** [comment]
Sure I see that latest changes are done on that branch. Btw why are we branching off from master? Are we planning to merge rucio 35 with master at some point?

**Comment by haozturk** [comment]
Here's the PR against rucio_35: https://github.com/dmwm/CMSRucio/pull/845

**Comment by dynamic-entropy** [comment]
Yes, will merge it to master soon.

**Comment by dynamic-entropy** [comment]
This will come with rucio_35 branch merging into main.

**Pull request closed without merging**